### PR TITLE
Participant tokens - remove / replace unsupportable from badges (conversion preparation)

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Participant tokens - remove / replace unsupportable from badges (conversion preparation)

Before
----------------------------------------
The drop down for event tokens offers 2 tokens for event title, start date, end date. One is 'standard' - ie `{event.title}` the other isn't ie `{participant.event_title}`

Also 2 tokens are offered that are hard to support and don't make sense in this context

`{participant.currency}` (does not return anything - the field is 'fee_currency' and doesn't make much sense in this context)
`{participant.note}` (this does work but I think we can drop support on this one as it is ambiguous if there are multiple notes, it's really hard to support, was exposed by accident and I think it's unlikely it is used).

After
----------------------------------------
Ambiguous tokens consolidated. Support dropped for the 2 problem tokens.

Technical Details
----------------------------------------
@seamuslee001 If we merge this then next steps are
1) add a participant token processor & switch this code to use it (date fields are the only issue I can see)
2) update pdf & email classes to use these (once cleanup is finished on those classes)

Comments
----------------------------------------
I added to the upgrade notes
https://lab.civicrm.org/-/ide/project/documentation/docs/sysadmin/tree/case/-/docs/upgrade/version-specific.md/

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
It turns out 2 more tokens will need to go to be supportable. They are in #21530 - I could add them to this but I'm kinda hoping this will be merged & I can simplify that

```
    '{participant.default_role_id}' => 'Default Role',
      '{participant.template_title}' => 'Event Template Title',
```

https://github.com/civicrm/civicrm-core/pull/21530/files#diff-d8f3ea846fa5593a3eec2f1e0c755a5bdade3580b5db78ff81a779340326a6e9L602-L603

File: CRM/Core/SelectValues.php [hunk 1]
```diff
@@ -595,14 +595,9 @@ public static function participantTokens(): array {
...
       '{participant.transferred_to_contact_id}' => 'Transferred to Contact ID',
       '{participant.participant_role}' => 'Participant Role (label)',
-      '{participant.event_title}' => 'Event Title',
-      '{participant.event_start_date}' => 'Event Start Date',
-      '{participant.event_end_date}' => 'Event End Date',
       '{participant.fee_label}' => 'Fee Label',
       '{participant.default_role_id}' => 'Default Role',
       '{participant.template_title}' => 'Event Template Title',
-      '{participant.currency}' => 'Currency',
-      '{participant.participant_note}' => 'Participant Note',
```
Comment (seamuslee001) [inline CRM/Core/SelectValues.php:605]: @eileenmcnaughton is there a replacement suggested for these 2?

Comment (seamuslee001) [reply CRM/Core/SelectValues.php:605]: ok I guess that is enough then

**Comment by seamuslee001** [comment]
This seems fine to me merging

**Pull request merged**